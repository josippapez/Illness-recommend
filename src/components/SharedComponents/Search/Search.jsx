import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

import './Search.scss';

const Search = props => {
  const [searchTerm, setSearchTerm] = useState('');
  const dispatch = useDispatch();

  return (
    <div className="search">
      <form
        className="search__form"
        onSubmit={e => {
          e.preventDefault();
          if (props.setSearchQuery) {
            if (searchTerm === '') {
              props.setSearchQuery(null);
              dispatch(props.fetchData());
            } else {
              props.setSearchQuery(searchTerm);
              dispatch(props.fetchDataByName(searchTerm));
            }
          }
        }}
      >
        <input
          type="text"
          disabled={props.disabled}
          placeholder="PretraÅ¾ivanje"
          className="search-input"
          value={searchTerm}
          onChange={e => setSearchTerm(e.target.value)}
        />
      </form>
      <small className="search-by-text">{props.searchingByInfo}</small>
    </div>
  );
};

Search.propTypes = {
  fetchDataByName: PropTypes.func.isRequired,
  fetchData: PropTypes.func.isRequired,
  setSearchQuery: PropTypes.func,
  searchingByInfo: PropTypes.string,
  disabled: PropTypes.bool,
};

export default Search;
