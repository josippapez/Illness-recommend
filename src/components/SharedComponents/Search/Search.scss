.navigation-setup:is(.darkTheme) {
  .search {
    .search-by-text {
      color: var(--white);
    }
    &__form {
      .search-input {
        background-color: var(--input-dark-background);
        background-image: url("../../../styles/assets/images/SearchIconWhite.svg");
        border: 1px solid transparent;
        color: var(--white);
        transition: all 0.2s ease-in-out;
      }
    }
  }
}

.search {
  width: 400px;
  min-height: 38px;
  @media screen and (max-width: 430px) {
    width: 100% !important;
  }
  .search-by-text {
    color: #5c5c5c;
    transition: all 0.2s ease-in-out;
  }
  &__form {
    display: flex;
    float: left;
    width: 100% !important;
    min-width: fit-content;
    position: relative;
    .search-input {
      transition: all 0.2s ease-in-out;
      height: 38px;
      width: 100%;
      padding: 0 45px 0 15px;
      background: var(--white) 0% 0% no-repeat padding-box;
      border: 1px solid #dededf;
      border-radius: 4px;
      background-image: url("../../../styles/assets/images/SearchIcon.svg");
      background-repeat: no-repeat;
      background-position: right 15px center;
      font: normal normal normal 14px "Lato, Regular";
      &:disabled {
        background: rgb(219, 218, 218);
      }
    }
  }
}
