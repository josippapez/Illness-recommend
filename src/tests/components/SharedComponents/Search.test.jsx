/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import Search from '../../../components/SharedComponents/Search/Search';

jest.mock('../../../store/actions');

const middlewares = [thunk];

const initialState = {};

const props = {
  fetchData: jest.fn(),
  setSearchQuery: jest.fn(),
};

const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

describe('Search', () => {
  const setSearchTerm = jest.fn();
  const handleClick = jest.spyOn(React, 'useState');
  handleClick.mockImplementation(searchTerm => [searchTerm, setSearchTerm]);

  it('renders correctly', () => {
    const component = mount(
      <Provider store={store}>
        <Search
          {...props}
          fetchDataByName={jest.fn().mockImplementation(() => ({
            type: 'test',
          }))}
        />
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });

  it('has the initial state empty string', () => {
    expect(setSearchTerm).toBeTruthy();
  });

  it('search content changes on input', () => {
    const component = mount(
      <Provider store={store}>
        <Search
          {...props}
          fetchDataByName={jest.fn().mockImplementation(() => ({
            type: 'test',
          }))}
        />
      </Provider>
    );
    component.find('.search-input').simulate('change', {
      target: { value: 'test' },
    });
    expect(component.find('.search-input').props().value).toEqual('test');
  });

  it('Dispatches search action when input is not null and enter is pressed', () => {
    const component = mount(
      <Provider store={store}>
        <Search
          {...props}
          fetchDataByName={jest.fn().mockImplementation(() => ({
            type: 'test',
          }))}
        />
      </Provider>
    );
    component.find('.search-input').simulate('change', {
      target: { value: 'test' },
    });
    expect(component.find('.search-input').props().value).toEqual('test');
    component.find('.search-input').simulate('submit');
  });

  it('Dispatches search action when input is null and enter is pressed', () => {
    const component = mount(
      <Provider store={store}>
        <Search
          {...props}
          fetchDataByName={jest.fn().mockImplementation(() => ({
            type: 'test',
          }))}
          fetchData={jest.fn().mockImplementation(() => ({
            type: 'test',
          }))}
        />
      </Provider>
    );
    component.find('.search-input').simulate('submit');
  });
});
