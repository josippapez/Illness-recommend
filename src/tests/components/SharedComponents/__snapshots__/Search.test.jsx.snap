// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `"<div class=\\"search\\"><form class=\\"search__form\\"><input type=\\"text\\" placeholder=\\"PretraÅ¾ivanje\\" class=\\"search-input\\" value=\\"\\"></form><small class=\\"search-by-text\\"></small></div>"`;
